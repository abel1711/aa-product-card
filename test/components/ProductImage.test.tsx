import React from 'react';
import renderer from 'react-test-renderer';
import ProductCard, { ProductImage } from '../../src/components';
import { product2 } from '../data/products';



describe('Pruebas al ProductImage', () => {

    test('debe de mostrar correctamente la img enviada por las props', () => {
        
        const wrapper = renderer.create(
            <ProductImage img='image-test'/>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('debe de mostrar la img del product', () => {
        
        const wrapper = renderer.create(
            <ProductCard product={ product2}>
                {
                    ()=>(
                        <ProductImage />
                    )
                }
            </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
})
