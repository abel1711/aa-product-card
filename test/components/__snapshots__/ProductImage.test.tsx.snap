// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al ProductImage debe de mostrar correctamente la img enviada por las props 1`] = `
<img
  alt="Product img"
  className="productImg undefined"
  src="image-test"
/>
`;

exports[`Pruebas al ProductImage debe de mostrar la img del product 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product img"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
